src/tests/ll_verification.py

```diff
@@ -723,6 +723,269 @@ def ll_ddi_adv_bv_21_c(transport, upperTester, lowerTester, trace):
+                    success = success and status == 0x43 # Limit Reached
+
+        # Check ADV_EXT_INDs
+        # AdvMode set to 00b; The ADV_EXT_IND PDU shall not include the SyncInfo, TxPower, ACAD, or AdvData
+        # fields. If advertising data was set, the ADV_EXT_IND PDU shall include the AuxPtr field;
+        # otherwise, the ADV_EXT_IND PDU *may* include the AuxPtr field. If the AuxPtr field is included,
+        # the ADV_EXT_IND PDU shall also include the ADI field with the SID set to the value used; otherwise that field shall not be included
+        for packet in packets.fetch(packet_filter=('ADV_EXT_IND')):
+            success = success and packet.payload['AdvMode'] == 0b00
+            success = success and 'SyncInfo' not in packet.payload
+            success = success and 'TxPower' not in packet.payload
+            success = success and 'ACAD' not in packet.payload
+            success = success and 'AD' not in packet.payload
+            if round.DataLength > 0:
+                success = success and 'AuxPtr' in packet.payload
+            if 'AuxPtr' in packet.payload:
+                success = success and 'ADI' in packet.payload
+                if 'ADI' in packet.payload:
+                    success = success and packet.payload['ADI'].SID == Sid
+
+        completeAdvDataFound = 0
+        # Check AUX_ADV_INDs
+        # AdvMode set to 00b; The AUX_ADV_IND PDU shall not include
+        # the SyncInfo, or TxPower fields. The AUX_ADV_IND PDU shall include the ADI field
+        # matching the ADI field from the ADV_EXT_IND. If the AUX_ADV_IND PDU does not contain all advertising data
+        # it shall include an AuxPtr field.
+        # The time between a PDU containing an AuxPtr field and the PDU to which it refers shall be greater than or equal to T_MAFS
+        for packet in packets.fetch(packet_filter=('AUX_ADV_IND')):
+            success = success and packet.payload['AdvMode'] == 0b00
+            success = success and 'SyncInfo' not in packet.payload
+            success = success and 'TxPower' not in packet.payload
+            success = success and 'ADI' in packet.payload
+            success = success and packet.payload['ADI'].SID == Sid
+            for superiorPacket in packet.payload['SuperiorPackets']:
+                success = success and packet.payload['ADI'].DID == superiorPacket.payload['ADI'].DID
+                success = success and packet.ts >= superiorPacket.ts + 300
+            if round.DataLength > 0:
+                if round.DataLength > len(packet.payload['AD']):
+                    success = success and 'AuxPtr' in packet.payload
+                else:
+                    # Check advertising data against input
+                    completeAdvDataFound += 1
+                    success = success and len(packet.payload['AD']) == round.DataLength
+                    for i in range(round.DataLength):
+                        success = success and packet.payload['AD'][i] == advData[i]
```

Comment:
Could this be just `success = success and packet.payload['AD'] == advData`?

Reply:
Unfortunately no, that doesn't work. AFAIK for that to work the two variables have to be pointing to the same underlying data; In this case it is not only two different sets of data, it is also different data types (slice of ints vs a `bytes` type)

Reply:
I had not realized that the types are different. `list(packet.payload['AD']) == list(advData)` would work, but that might be less efficient than the present code.